examples/se_2_3_localization.cpp

```diff
@@ -0,0 +1,452 @@
+ *  to estimate the state, using a Lie-based error-state Kalman filter.
+ *
+ *  This file has plain code with only one main() function.
+ *  There are no function calls other than those involving `manif`.
+ *
+ *  Printing simulated state and estimated state together
+ *  with an unfiltered state (i.e. without Kalman corrections)
+ *  allows for evaluating the quality of the estimates.
+ *
+ * A side note: Besides the approach described here in this illustration example,
+ * there are other interesting works like the paper,
+ * The Invariant Extended Kalman filter as a stable observer (https://arxiv.org/pdf/1410.1465.pdf)
+ * which assume a specific structure for the system propagation dynamics "f(X_t, u)" (group affine dynamics)
+ * that simplifies the covariance propagation and enables error dynamics with stronger convergence properties.
+ *
+ */
+
+#include "manif/SE_2_3.h"
+
+#include <Eigen/Dense>
+
+#include <vector>
+#include <iostream>
+#include <iomanip>
+
+using std::cout;
+using std::endl;
+
+using namespace Eigen;
+
+typedef Array<double, 3, 1> Array3d;
+typedef Array<double, 6, 1> Array6d;
+typedef Matrix<double, 6, 1> Vector6d;
+typedef Matrix<double, 6, 6> Matrix6d;
+typedef Array<double, 9, 1> Array9d;
+typedef Matrix<double, 9, 1> Vector9d;
+typedef Matrix<double, 9, 9> Matrix9d;
+
+int main()
+{
+    // START CONFIGURATION
+    //
+    //
+    const int NUMBER_OF_LMKS_TO_MEASURE = 5;
+
+    // Define the robot extended pose element and its covariance
+    manif::SE_2_3d X, X_simulation, X_unfiltered;
+    Matrix9d    P;
+
+    X_simulation.setIdentity();
+    X.setIdentity();
+    X_unfiltered.setIdentity();
+    P.setZero();
+    P.block<3, 3>(0, 0) = 0.001*Eigen::Matrix3d::Identity();
+    P.block<3, 3>(3, 3) = 0.01*Eigen::Matrix3d::Identity();
+    P.block<3, 3>(6, 6) = 0.001*Eigen::Matrix3d::Identity();
+
+    // acceleration due to gravity in world frame
+    Vector3d g;
+    g << 0, 0, -9.80665;
+    const double dt = 0.01;
+
+    // IMU measurements in IMU frame
+    Vector3d alpha, alpha_const, omega, alpha_prev, omega_prev;
+    alpha_const << 0.1, 0.01, 0.1; // constant acceleration in IMU frame without gravity compensation
+    omega << 0.01, 0.1, 0; // constant angular velocity about x- and y-direction in IMU frame
+
+    // Previous IMU measurements in IMU frame initialized to values expected when stationary
+    alpha_prev = alpha = alpha_const - (X_simulation.rotation()).transpose()*g;
+    omega_prev << 0, 0, 0;
+
+    // Define a control vector and its noise and covariance
+    manif::SE_2_3Tangentd  u_simu, u_est, u_unfilt;
+    Vector9d            u_nom, u_noisy, u_noise;
+    Array9d             u_sigmas;
+    Matrix9d            U;
+
+    u_sigmas << 0.0, 0.0, 0.0, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01;
+    U        = (u_sigmas * u_sigmas).matrix().asDiagonal();
+
+    // Declare the Jacobians of the motion wrt robot and control
+    manif::SE_2_3d::Jacobian slantFk, Fk, J_x, J_u;
```

Comment:
Use `J_y_x = dy/dx` consistently for each one of the functions. then use the chain rule as in e.g. `J_z_x = J_z_y*J_y_x` or even `J_z_x = J_z_y*J_y_x + J_z_u*J_u_x`.

Reply:
I think doing these easy changes will do. I am checking below for the Jacobian of the input vector to see if we can also improve the notation.